StoreMessageContentActivity/handler.ts

```diff
@@ -53,13 +67,107 @@ export type StoreMessageContentActivityResult = t.TypeOf<
   typeof StoreMessageContentActivityResult
 >;
 
+export const ServicePreferenceError = t.interface({
+  kind: t.keyof({ ERROR: null, LEGACY: null }),
+  message: t.string
+});
+
+export type ServicePreferenceError = t.TypeOf<typeof ServicePreferenceError>;
+
+export type ServicePreferenceValueOrError = (params: {
+  readonly serviceId: NonEmptyString;
+  readonly fiscalCode: FiscalCode;
+  readonly userServicePreferencesMode: ServicesPreferencesMode;
+  readonly userServicePreferencesVersion: number;
+}) => TaskEither<
+  ServicePreferenceError,
+  ReadonlyArray<BlockedInboxOrChannelEnum>
+>;
+
+type ServicePreferencesValues = Omit<
+  ServicePreference,
+  "serviceId" | "fiscalCode" | "settingsVersion"
+>;
+
+const channelToBlockedInboxOrChannelEnum: {
+  readonly [key in keyof ServicePreferencesValues]: BlockedInboxOrChannelEnum;
+} = {
+  isEmailEnabled: BlockedInboxOrChannelEnum.EMAIL,
+  isInboxEnabled: BlockedInboxOrChannelEnum.INBOX,
+  isWebhookEnabled: BlockedInboxOrChannelEnum.WEBHOOK
+};
+
+const servicePreferenceToBlockedInboxOrChannels: (
+  servicePreference: ServicePreference
+) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference =>
+  Object.entries(servicePreference)
+    // take only attributes of ServicePreferencesValues
```

Comment:
```suggestion
) => ReadonlyArray<BlockedInboxOrChannelEnum> = servicePreference =>
  /**
   * Reduce the complexity of User's preferences into an array of BlockedInboxOrChannelEnum
   * In case a preference is set to false, it is translated to proper BlockedInboxOrChannelEnum
   * and added to returned array.
   * By adding a `channelToBlockedInboxOrChannelEnum` map we are prepared to handle new
   * service preferences
   */
  Object.entries(servicePreference)
    // take only attributes of ServicePreferencesValues
```